File: Python & Ruby/Multiple_Inheritance/mi.py
class C1():
  def c1_m(self):
    print("c1_m")
  def m(self):
    print("c1 m")

class C2():
  def c2_m(self):
    print("c2_m")
  def m(self):
    print("c2 m")

class C3(C1, C2):
  pass

c = C3()
c.c1_m()
c.c2_m()
c.m()
print(C3.__mro__)

# class CalMultiply():
#   def multiply(self):
#     return self.v1 * self.v2

# class CalDivied():
#   def devide(self):
#     return self.v1 / self.v2

# class Cal(CalMultiply, CalDivied):
#   def __init__(self, v1, v2):
#     if isinstance(v1, int):
#       self.v1 = v1
#     if isinstance(v2, int):
#       self.v2 = v2
#   def add(self):
#     return self.v1 + self.v2
#   def subtract(self):
#     return self.v1 - self.v2
#   def setV1(self, v):
#     if isinstance(v, int):
#       self.v1 = v
#   def getV1(self, v):
#     return self.v1

# c = Cal(100, 10)
# print(c.add())
# print(c.subtract())
# print(c.multiply())
# print(c.devide())

File: Python & Ruby/Multiple_Inheritance/mi.rb
# module M1
#   def m1_m
#     p "m1_m"
#   end
# end
# module M2
#   def m2_m
#     p "m2_m"
#   end
# end

# class C
#   include M1, M2
# end
# c = C.new()
# c.m1_m
# c.m2_m

module Multiply
  def multiply()
    return @v1 * @v2
  end
end

module Devide
  def devide()
    return @v1 / @v2
  end
end

class Cal
  include Multiply, Devide
  def initialize(v1, v2)
    @v1 = v1
    @v2 = v2
  end
  def add()
    return @v1 + @v2
  end
  def subtract()
    return @v1 - @v2
  end
end

c = Cal.new(100, 10)
p c.add()
p c.subtract()
p c.multiply()
p c.devide()